test/built-ins/RegExp/lookbehind.js

```diff
@@ -0,0 +1,169 @@
+assert.sameValue(null, "ab".match(/(..)(?<=\1\1\1)/));
+assert.sameValue(null, "abb".match(/(..)(?<=\1\1\1)/));
+assert.sameValue(null, "aabb".match(/(..)(?<=\1\1\1)/));
+assert.sameValue(null, "abab".match(/(..)(?<=\1\1\1)/));
+assert.sameValue(null, "fabxbab".match(/(..)(?<=\1\1\1)/));
+assert.sameValue(null, "faxabab".match(/(..)(?<=\1\1\1)/));
+assert.compareArray(["ab", "ab"], "fababab".match(/(..)(?<=\1\1\1)/));
+
+// Back references to captures inside the lookbehind.
+assert.compareArray(["d", "C"], "abcCd".match(/(?<=\1(\w))d/i));
+assert.compareArray(["d", "x"], "abxxd".match(/(?<=\1([abx]))d/));
+assert.compareArray(["c", "ab"], "ababc".match(/(?<=\1(\w+))c/));
+assert.compareArray(["c", "b"], "ababbc".match(/(?<=\1(\w+))c/));
+assert.sameValue(null, "ababdc".match(/(?<=\1(\w+))c/));
+assert.compareArray(["c", "abab"], "ababc".match(/(?<=(\w+)\1)c/));
+
+// Alternations are tried left to right,
+// and we do not backtrack into a lookbehind.
+assert.compareArray(["xabcd", "cd", ""], "xabcd".match(/.*(?<=(..|...|....))(.*)/));
+assert.compareArray(["xabcd", "bcd", ""], "xabcd".match(/.*(?<=(xx|...|....))(.*)/));
+assert.compareArray(["xxabcd", "bcd", ""], "xxabcd".match(/.*(?<=(xx|...))(.*)/));
+assert.compareArray(["xxabcd", "xx", "abcd"], "xxabcd".match(/.*(?<=(xx|xxx))(.*)/));
+
+// We do not backtrack into a lookbehind.
+// The lookbehind captures "abc" so that \1 does not match. We do not backtrack
+// to capture only "bc" in the lookbehind.
+assert.sameValue(null, "abcdbc".match(/(?<=([abc]+)).\1/));
+
+// Greedy loop.
+assert.compareArray(["c", "bbbbbb"], "abbbbbbc".match(/(?<=(b+))c/));
+assert.compareArray(["c", "b1234"], "ab1234c".match(/(?<=(b\d+))c/));
+assert.compareArray(["c", "b12b23b34"], "ab12b23b34c".match(/(?<=((?:b\d{2})+))c/));
+
+// Sticky
+var re1 = /(?<=^(\w+))def/g;
+assert.compareArray(["def", "abc"], re1.exec("abcdefdef"));
+assert.compareArray(["def", "abcdef"], re1.exec("abcdefdef"));
+var re2 = /\Bdef/g;
+assert.compareArray(["def"], re2.exec("abcdefdef"));
+assert.compareArray(["def"], re2.exec("abcdefdef"));
+
+// Misc
+assert.sameValue(null, "abcdef".match(/(?<=$abc)def/));
+assert.compareArray(["foo"], "foo".match(/^foo(?<=foo)$/));
+assert.compareArray(["foo"], "foo".match(/^f.o(?<=foo)$/));
+assert.sameValue(null, "fno".match(/^f.o(?<=foo)$/));
+assert.sameValue(null, "foo".match(/^foo(?<!foo)$/));
+assert.sameValue(null, "foo".match(/^f.o(?<!foo)$/));
+assert.compareArray(["fno"], "fno".match(/^f.o(?<!foo)$/));
+assert.compareArray(["foooo"], "foooo".match(/^foooo(?<=fo+)$/));
+assert.compareArray(["foooo"], "foooo".match(/^foooo(?<=fo*)$/));
+assert.compareArray(["abc", "abc"], /(abc\1)/.exec("abc"));
+assert.compareArray(["abc", "abc"], /(abc\1)/.exec("abc\u1234"));
+assert.compareArray(["abc", "abc"], /(abc\1)/i.exec("abc"));
+assert.compareArray(["abc", "abc"], /(abc\1)/i.exec("abc\u1234"));
+var oob_subject = "abcdefghijklmnabcdefghijklmn".substr(14);
```

Comment:
I only need some sort of description like a comment - it's better in a separate file - otherwise this can be removed in any further maintenance because it's implementation specific and I can't observe the behavior from the specs.